src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -92,6 +95,11 @@ public void RegisterMissingWorkloads(Guid projectGuid, ConfiguredProject project
 
         public void RegisterMissingSdkRuntimeComponentId(Guid projectGuid, ConfiguredProject project, string runtimeComponentId)
         {
+            if (!_missingRuntimesRegistered.TryAdd(runtimeComponentId, 0))
+            {
+                return;
+            }
+
```

Comment:
Do we even need to track this? What is the downside to calling this method more than once.

As it stands, this early return means we only clean up the first project to call for a given `runtimeComponentId`.

Is it important to store all project GUIDs here? `UnregisterProjectConfiguration` might also need to be called for more than just the first caller.

Reply:
We want to avoid calling the Setup apis.